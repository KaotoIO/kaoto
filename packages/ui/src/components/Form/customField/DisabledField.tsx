import { Card, CardBody, CardTitle } from '@patternfly/react-core';
import { ConnectedFieldProps, connectField } from 'uniforms';
import { CustomExpandableSection } from './CustomExpandableSection';

interface CustomStepsFieldProps {
  'data-testid': string;
  [key: string]: string;
}

export const DisabledField = connectField((props: ConnectedFieldProps<CustomStepsFieldProps>) => {
  return (
    <Card>
      <CardTitle>{props.label}</CardTitle>
      <CardBody>
        <p>Configuring this field is not yet supported</p>

        <CustomExpandableSection groupName={props.name}>
          <code>
            <pre>{JSON.stringify(props, null, 2)}</pre>
          </code>
        </CustomExpandableSection>
      </CardBody>
    </Card>
  );
});
