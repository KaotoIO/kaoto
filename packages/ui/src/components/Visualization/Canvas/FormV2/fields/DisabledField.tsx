import { Card, CardBody, CardTitle } from '@patternfly/react-core';
import { FunctionComponent } from 'react';
import { IDataTestID } from '../../../../../models';
import { FieldProps } from '../typings';
import { CustomExpandableSection } from '../../../../Form/customField/CustomExpandableSection';

export const DisabledField: FunctionComponent<IDataTestID & FieldProps> = (props) => {
  return (
    <Card>
      <CardTitle>{props.propName}</CardTitle>
      <CardBody>
        <p>Configuring this field is not yet supported</p>

        <CustomExpandableSection groupName={props.propName}>
          <code>
            <pre>{JSON.stringify(props, null, 2)}</pre>
          </code>
        </CustomExpandableSection>
      </CardBody>
    </Card>
  );
};
