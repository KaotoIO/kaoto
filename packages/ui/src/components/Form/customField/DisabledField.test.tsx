jest.mock('uniforms', () => {
  const uniforms = jest.requireActual('uniforms');

  return {
    ...uniforms,
    connectField: (fn: () => void) => fn,
  };
});
import { render } from '@testing-library/react';
import { DisabledField } from './DisabledField';

describe('DisabledField', () => {
  it('should render', () => {
    const { container } = render(<DisabledField id="disabled-field-id" name="test" label="Disabled field label" />);

    expect(container).toMatchSnapshot();
  });
});
