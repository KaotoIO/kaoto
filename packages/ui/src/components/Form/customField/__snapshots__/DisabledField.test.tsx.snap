// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisabledField should render 1`] = `
<div>
  <div
    class="pf-v5-c-card"
    data-ouia-component-id="OUIA-Generated-Card-1"
    data-ouia-component-type="PF5/Card"
    data-ouia-safe="true"
    id=""
  >
    <div
      class="pf-v5-c-card__title"
    >
      <div
        class="pf-v5-c-card__title-text"
      >
        Disabled field label
      </div>
    </div>
    <div
      class="pf-v5-c-card__body"
    >
      <p>
        Configuring this field is not yet supported
      </p>
      <div
        class="pf-v5-c-expandable-section"
      >
        <button
          aria-controls="expandable-section-content"
          aria-expanded="false"
          class="pf-v5-c-expandable-section__toggle"
          id="expandable-section-toggle-test"
          type="button"
        >
          <span
            class="pf-v5-c-expandable-section__toggle-icon"
          >
            <svg
              aria-hidden="true"
              class="pf-v5-svg"
              fill="currentColor"
              height="1em"
              role="img"
              viewBox="0 0 256 512"
              width="1em"
            >
              <path
                d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
              />
            </svg>
          </span>
          <span
            class="pf-v5-c-expandable-section__toggle-text"
          >
            Test properties
          </span>
        </button>
        <div
          aria-labelledby="expandable-section-toggle-test"
          class="pf-v5-c-expandable-section__content"
          hidden=""
          id="expandable-section-content"
          role="region"
        >
          <code>
            <pre>
              {
  "id": "disabled-field-id",
  "name": "test",
  "label": "Disabled field label"
}
            </pre>
          </code>
        </div>
      </div>
    </div>
  </div>
</div>
`;
